tests/integration/test_nanquantiles.py

```diff
@@ -0,0 +1,98 @@
+# Copyright 2022 NVIDIA Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+
+import numpy as np
+import pytest
+from legate.core import LEGATE_MAX_DIM
+from utils.comparisons import allclose
+
+import cunumeric as num
+
+ALL_METHODS = (
+    "inverted_cdf",
+    "averaged_inverted_cdf",
+    "closest_observation",
+    "interpolated_inverted_cdf",
+    "hazen",
+    "weibull",
+    "linear",
+    "median_unbiased",
+    "normal_unbiased",
+    "lower",
+    "higher",
+    "midpoint",
+    "nearest",
+)
+
+
+@pytest.mark.parametrize("str_method", ALL_METHODS)
+@pytest.mark.parametrize("axes", (0, 1))
+@pytest.mark.parametrize(
+    "qin_arr", (0.5, [0.001, 0.37, 0.42, 0.67, 0.83, 0.99, 0.39, 0.49, 0.5])
+)
+@pytest.mark.parametrize("keepdims", (False,))
+@pytest.mark.parametrize("overwrite_input", (False,))
+def test_multi_axes(str_method, axes, qin_arr, keepdims, overwrite_input):
+    eps = 1.0e-8
+    arr = np.ndarray(
+        shape=(3, 4),
+        buffer=np.array(
+            [1.0, 2, 2, 40, 1, np.nan, 2, 1, 0, 10, np.nan, np.nan]
+        ),
+        dtype=float,
+    )
+
+    if num.isscalar(qin_arr):
+        qs_arr = qin_arr
+    else:
+        qs_arr = np.array(qin_arr)
+
+    # cunumeric:
+    # print("cunumeric axis = %d:"%(axis))
+    q_out = num.nanquantile(
+        arr,
+        qs_arr,
+        axis=axes,
+        method=str_method,
+        keepdims=keepdims,
+        overwrite_input=overwrite_input,
+    )
+    # print(q_out)
+
+    # np:
+    # print("numpy axis = %d:"%(axis))
+    np_q_out = np.nanquantile(
+        arr,
+        qs_arr,
+        axis=axes,
+        method=str_method,
+        keepdims=keepdims,
+        overwrite_input=overwrite_input,
+    )
+    # print(np_q_out)
+
+    assert q_out.shape == np_q_out.shape
+    assert q_out.dtype == np_q_out.dtype
+
+    assert allclose(np_q_out, q_out, atol=eps)
+
+
+if __name__ == "__main__":
+    import sys
+
+    np.random.seed(12345)
+
```

Comment:
Shouldn't be necessary anymore; Legate-Core will do this proactively.
```suggestion
```

Reply:
Fixed.